azext_edge/tests/edge/support/conftest.py

```diff
@@ -31,17 +31,21 @@ def add_pod_to_mocked_pods(
         spec = V1PodSpec(containers=[V1Container(name=container_name)])
         pod = V1Pod(metadata=V1ObjectMeta(namespace=namespace, name=pod_name), spec=spec)
 
-        if mock_init_containers:
+        if mock_init_container:
             pod.spec.init_containers = [V1Container(name="mock-init-container")]
-            expected_pod_map[namespace][pod_name]["mock-init-container"] = mock_log
+            expected_pod_map[namespace][pod_name] = {"mock-init-container": mock_log}
```

Comment:
make sure you are not losing values here (since this is overwriting a dict rather than value

Reply:
yeah I double checked, at this point pod_name under expected_pod_map[namespace] is not yet exists